experimental/packages/opentelemetry-browser-detector/src/util.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {Suite} from 'mocha';
+import * as assert from 'assert';
+import {BROWSER_ATTRIBUTES} from './types';
+import {Resource} from '@opentelemetry/resources';
+
+export function describeBrowser(title: string, fn: (this: Suite) => void) {
+  title = `Browser: ${title}`;
+  if (typeof process === 'object' && process.release?.name === 'node') {
+    return describe.skip(title, fn);
+  }
+  return describe(title, fn);
+}
+
+export const assertResource = (
+  resource: Resource,
+  validations: {
+    platform?: string,
+    brands?: string[],
+    mobile?: boolean,
+    user_agent?: string
+  }
+) => {
+  if (validations.platform) {
+    assert.strictEqual(
+      resource.attributes[BROWSER_ATTRIBUTES.PLATFORM],
+      validations.platform
+    );
+  }
+  if (validations.brands) {
+    assert.ok(Array.isArray(resource.attributes[BROWSER_ATTRIBUTES.BRANDS]));
+    assert.strictEqual(
+      (resource.attributes[BROWSER_ATTRIBUTES.BRANDS] as Array<string>).join('.'),
```

Comment:
Array literal types can be more straightforward when the type parameter is a simple type:

```suggestion
      (resource.attributes[BROWSER_ATTRIBUTES.BRANDS] as string[]).join('.'),
```